04-onion-routing.md

```diff
@@ -238,6 +240,10 @@ The requirements ensure consistency in responding to an unexpected
 `outgoing_cltv_value`, whether it is the final node or not, to avoid
 leaking its position in the route.
 
+`attributable_error` is a marker record that specifies that the failure message
+in the `update_fail_htlc` response should be structured as an attributable
+error. For the legacy format, this record should not be set.
+
 ### Requirements
```

Comment:
We need to add requirements for the reader.
It may also be useful to create a new failure code for when the downstream node sent us a legacy error.

Reply:
Requirement added for the reader.

Reply:
>It may also be useful to create a new failure code for when the downstream node sent us a legacy error.

I don't think we can know this, can we?

Reply:
We can't always know for sure but the size of the packet we receive should be a good indicator, legacy error being much shorter than attributable errors.
What do you do if the downstream node returns a legacy error (or any error that's too short)? There are no payloads and hmacs to shift.

Reply:
Ah I see. In https://github.com/lightningnetwork/lnd/pull/7139, a properly-structured all-zeroes (attributable) failure message is returned in this case. With that, the sender at least has an indication of the offending node's position, but the message itself is not defined. 

Good suggestion to make that a specific error.